python/lsst/pipe/tasks/cosmicRayPostDiff.py

```diff
@@ -0,0 +1,143 @@
+
+
+__all__ = ["CosmicRayPostDiffConfig", "CosmicRayPostDiffTask"]
+
+
+class CosmicRayPostDiffConnections(pipeBase.PipelineTaskConnections,
+                                   dimensions=("instrument", "visit", "detector"),
+                                   defaultTemplates={"coaddName": "deep",
+                                                     "fakesType": ""}):
+    exposure = cT.Input(
+        doc="Input difference image",
+        name="{fakesType}{coaddName}Diff_differenceExp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    repaired = cT.Output(
+        doc="Output subtracted image after CR run.",
+        name="{fakesType}{coaddName}Diff_repairedExp",
+        storageClass="ExposureF",
+        dimensions=("instrument", "visit", "detector"),
+    )
+
+    def adjustQuantum(self, inputs, outputs, label, dataId):
+        # Docstring inherited from PipelineTaskConnections
+        try:
+            return super().adjustQuantum(inputs, outputs, label, dataId)
+        except pipeBase.ScalarError as err:
+            raise pipeBase.ScalarError(
+                "CosmicRayPostDiffTask can at present only be run on visits that are associated with "
+                "exactly one exposure.  Either this is not a valid exposure for this pipeline, or the "
+                "snap-combination step you probably want hasn't been configured to run between ISR and "
+                "this task (as of this writing, that would be because it hasn't been implemented yet)."
+            ) from err
+
+
+class CosmicRayPostDiffConfig(pipeBase.PipelineTaskConfig,
+                              pipelineConnections=CosmicRayPostDiffConnections):
+
+    """Config for CosmicRayPostDiffTask"""
+    repair = pexConfig.ConfigurableField(
+        target=RepairTask,
+        doc="Remove cosmic rays",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+
+
+class CosmicRayPostDiffTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    r"""Measure bright sources and use this to estimate background and PSF of an exposure
+
+    @anchor CosmicRayPostDiffTask_
+
+    @section pipe_tasks_cosmicRayPostDiff_Contents  Contents
+
+     - @ref pipe_tasks_cosmicRayPostDiff_Purpose
+     - @ref pipe_tasks_cosmicRayPostDiff_Initialize
+     - @ref pipe_tasks_cosmicRayPostDiff_IO
+     - @ref pipe_tasks_cosmicRayPostDiff_Config
+
+
+    @section pipe_tasks_cosmicRayPostDiff_Purpose  Description
+
+    Given an output image from image differencing:
+    - detect and repair cosmic rays
+    - At the moment this task does NOT recompute the PSF or re-do source detection.
+
+    @section pipe_tasks_cosmicRayPostDiff_Initialize  Task initialisation
+
+    @copydoc \_\_init\_\_
+
+    @section pipe_tasks_cosmicRayPostDiff_IO  Invoking the Task
+
+    Call the `run` method.
+
+    @section pipe_tasks_cosmicRayPostDiff_Config  Configuration parameters
+
+    See @ref CosmicRayPostDiffConfig
```

Comment:
This uses the old formatting. I think it could be simplified to:

```
"""Detect and repair cosmic rays on an image difference.

      Given an output image from image differencing:
      - detect and repair cosmic rays
      - At the moment this task does NOT recompute the PSF or re-do source detection.
 """
```